protocol.go

```diff
@@ -40,7 +40,7 @@ const (
 	headerUserAgent   = "User-Agent"
 	headerTrailer     = "Trailer"
 
-	discardLimit = 1024 * 1024 * 4 // 4MiB
+	discardLimit = 1024 * 1024 * 8 // 8MiB
```

Comment:
Test was failing now that discard is used. I've upped this limit to avoid changing the test.

Reply:
Which test? I think it might be better to change the test. While 8mb is not a big number, it can be bad combined with a slow client. And until we solve the "read timeout" thing (discussed in separate threads), we should be more conservative. 4mb is already generous since most RPC payloads are single digit kilobytes, so doubling this seems like the wrong play.